nixos/tests/matrix-synapse.nix

```diff
@@ -0,0 +1,35 @@
+import ./make-test.nix ({ pkgs, ... } :
+
+with pkgs.lib;
+
+let
+  server = args: {
+    services.matrix-synapse.enable = true;
+  };
+
+in {
+  name = "matrix-synapse";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ corngood ];
+  };
+
+  nodes = {
+    server_postgres = server;
+
+    server_sqlite = args: server args // {
+      system.stateVersion = "17.09";
```

Comment:
Wouldn't it be better for this to just explicitly turn on sqlite? Who knows what other side effects this specific stateVersion might come to have over time.

Reply:
Yeah, I guess I originally intended to verify that it was using the correct db for each stateVersion, but I didn't actually test for it.  I suppose the stateVersion logic is pretty simple, and it would be more intuitive to explicitly test the backends.